msal-go/index.md

```diff
@@ -0,0 +1,106 @@
+
+1. MSAL separates [public and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1). So, you would create an instance of a `PublicClientApplication` and `ConfidentialClientApplication` and use this throughout the lifetime of your application.
+
+   - Initializing a public client:
+
+    ```go
+    publicClientApp, err := public.New("client_id", public.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+   - Initializing a confidential client:
+
+    ```go
+    // Initializing the client credential
+    cred, err := confidential.NewCredFromSecret("client_secret")
+    if err != nil {
+        return nil, fmt.Errorf("could not create a cred from a secret: %w", err)
+    }
+    confidentialClientApp, err := confidential.New("client_id", cred, confidential.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+1. MSAL comes packaged with an in-memory cache. Utilizing the cache is optional, but we would highly recommend it.
+
+    ```go
+    var userAccount public.Account
+    accounts := publicClientApp.Accounts()
+    if len(accounts) > 0 {
+        // Assuming the user wanted the first account
+        userAccount = accounts[0]
+        // found a cached account, now see if an applicable token has been cached
+        result, err := publicClientApp.AcquireTokenSilent(context.Background(), []string{"your_scope"}, public.WithSilentAccount(userAccount))
+        accessToken := result.AccessToken
+    }
+    ```
+
+1. If there is no suitable token in the cache, or you choose to skip this step, now we can send a request to AAD to obtain a token.
+
+    ```go
+    result, err := publicClientApp.AcquireTokenByOneofTheActualMethods([]string{"your_scope"}, ...(other parameters depending on the function))
+    if err != nil {
+        log.Fatal(err)
+    }
+    accessToken := result.AccessToken
+    ```
+
+You can view the [dev apps](https://github.com/AzureAD/microsoft-authentication-library-for-go/tree/dev/apps/tests/devapps) on how to use MSAL Go with various application types in various scenarios. For more detailed information, please refer to the [wiki](https://github.com/AzureAD/microsoft-authentication-library-for-go/wiki).
+
+## Releases
+
+For a full list of library releases, refer to the [Releases](https://github.com/AzureAD/microsoft-authentication-library-for-go/releases) section in the library source code repository.
+
+## Community help and support
+
+We use [Stack Overflow](https://stackoverflow.com/questions/tagged/azure-ad-msal) to work with the community on supporting Azure Active Directory and its SDKs, including this one! We highly recommend you ask your questions on Stack Overflow. You can also browse existing questions to see if someone has encountered the problem before. Please use the `azure-ad-msal` tag when asking your questions.
```

Comment:
Do we really monitor Stack Overflow?

Reply:
I believe there are some support engineers from Microsoft monitoring StackOverflow. Not sure how often. MSAL team here historically does not monitor StackOverflow. That being said, occasionally when I receive 1:1 questions on Teams, and if the question is generic, I suggest the question asker to post their question on StackOverflow, and paste me the link if it got unanswered after a day or two.

TL;DR. I would still suggest keep this StackOverflow paragraph.

Reply:
I am monitoring Stack Overflow @bgavrilMS @rayluo 😄